x/lazyledgerapp/client/builder.go

```diff
@@ -0,0 +1,172 @@
+package client
+
+import (
+	"context"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/lazyledger/lazyledger-app/app/params"
+	"google.golang.org/grpc"
+)
+
+// Builder wraps the standard sdkclient.TxBuilder add a few quality of life methods along
+// with keeping track of the account number to produce transactions
+type Builder struct {
+	sdkclient.TxBuilder
+	encCfg               params.EncodingConfig
+	address              sdktypes.Address
+	chainID              string
+	accountNum, sequence uint64
+}
+
+func NewBuilder(address sdktypes.Address, chainID string) *Builder {
+	encCfg := params.RegisterAccountInterface(params.MakeEncodingConfig())
+	txBuilder := encCfg.TxConfig.NewTxBuilder()
+	return &Builder{
+		TxBuilder: txBuilder,
+		encCfg:    encCfg,
+		address:   address,
+		chainID:   chainID,
+	}
+}
+
+// UpdateAccountNumber queries the applicaiton to find the latest account number and
+// sequence, updating the respective internal fields
+func (b *Builder) UpdateAccountNumber(ctx context.Context, conn *grpc.ClientConn) error {
```

Comment:
Do we need to export this function?

Reply:
We actually do need to export this one, as the account number needs to be updated for the signed transaction to be valid. Here's an example in the [optimint PR](https://github.com/evan-forbes/optimint/blob/de5e447777b8541adff6bbc09a8a694e0e4f0fae/da/lazyledger/lazyledger.go#L110) 

However, that was clearly not explicit enough, and an extra line of documentation https://github.com/lazyledger/lazyledger-app/pull/94/commits/b5c730019f1d033beb20ba9bb13e69ea372a0c80

Reply:
And TestBroadcastPayForMessage also kinda documents that.